doc/modules/lmnn.rst

```diff
@@ -0,0 +1,120 @@
+.. _lmnn:
+
+=============================
+Large Margin Nearest Neighbor
+=============================
+
+.. sectionauthor:: John Chiotellis <johnyc.code@gmail.com>
+
+.. currentmodule:: sklearn.neighbors
+
+Large Margin Nearest Neighbor (LMNN, :class:`LargeMarginNearestNeighbor`) is a distance metric learning algorithm which aims to improve the accuracy of nearest neighbors classification compared to the standard Euclidean distance.
+
```

Comment:
It would be really nice to have a figure here that gives an intuition, for people that do a quick reading of the documentation.

Reply:
Done.